lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -157,15 +157,10 @@ def parse(body)
       end
 
       def success_from(response)
-        return true if (response.is_a?(Hash) && response[:next_step]) || response == 204
-
-        return false unless %w(PENDING EXPIRED PAID).include?(response[:status])
-
-        response[:transactions].each do |transaction|
-          return false unless %w(PENDING CANCELLED SUCCESSFUL).include?(transaction.symbolize_keys[:status])
-        end
-
-        true
+        (response.is_a?(Hash) && response[:next_step]) ||
+          response == 204 ||
+          %w(PENDING EXPIRED PAID).include?(response[:status]) ||
```

Comment:
❓ I want to be sure.
Mean this that, if the value is PENDING, EXPIRED, or PAID, or if all the response transactions have PENDING CANCELLED or SUCCESSFUL this will be a successful transaction?

EXPIRED, and CANCELLED are considered successful transactions?

Reply:
EXPIRED and in the transactions object PENDING and CANCELLED was removed, these states were necessary when canceling a transaction that had been left in a pending state, but this behavior is not supported by Spreedly